devito/petsc/iet/passes.py

```diff
@@ -0,0 +1,366 @@
+from devito.passes.iet.engine import iet_pass
+from devito.ir.iet import (FindNodes, Call,
+                           Transformer, FindSymbols,
+                           MapNodes, Iteration, Callable, Callback, List, Uxreplace,
+                           Definition, BlankLine, PointerCast)
+from devito.petsc.types import (PetscMPIInt, PETScStruct, DM, Mat,
+                                Vec, KSP, PC, SNES, PetscErrorCode, PETScArray)
+from devito.symbolics import Byref, Macro, FieldFromPointer
+from devito.petsc.iet.nodes import MatVecAction, LinearSolverExpression
+from devito.petsc.utils import linear_solver_mapper
+import cgen as c
+
+
+@iet_pass
+def lower_petsc(iet, **kwargs):
+
+    # Check if PETScSolve was used.
```

Comment:
nitpicking: we use full stops within docstrings, but not at the end of simple comments